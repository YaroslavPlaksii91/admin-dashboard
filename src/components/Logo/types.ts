import { LOGO_TYPES } from './constants';

export type LogoProps = {
  type?: (typeof LOGO_TYPES)[keyof typeof LOGO_TYPES];
};
