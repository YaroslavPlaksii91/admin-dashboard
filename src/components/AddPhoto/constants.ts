export const IMAGE_SIZE = Object.freeze({
  SMALL: 44,
  BIG: 60,
});
