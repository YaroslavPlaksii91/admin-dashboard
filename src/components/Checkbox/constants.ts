export const CHECKBOX_TEST_ID = 'checkbox';
