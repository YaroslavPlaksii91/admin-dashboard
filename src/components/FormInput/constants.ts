export const TOGGLE_BUTTON_TEST_ID = 'toggleButton';
