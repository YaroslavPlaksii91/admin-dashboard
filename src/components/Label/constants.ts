export const LABEL_TYPES = Object.freeze({
  ROUND: 'round',
  SQUARE: 'square',
});

export const LABEL_COLORS = Object.freeze({
  DEFAULT: 'default',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error',
});
