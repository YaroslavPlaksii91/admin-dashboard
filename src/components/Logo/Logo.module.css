.wrapperVertical {
  margin-bottom: 32px;
}

.wrapperHorizontal {
  margin-bottom: 59px;
}

.wrapperVertical .iconVertical {
  font-size: 48px;
  margin-bottom: 12px;
}

.wrapperHorizontal .iconHorizontal {
  font-size: 32px;
  margin-right: 12px;
}

.wrapperHorizontal .link {
  display: flex;
  align-items: center;
  padding-left: 32px;
}
