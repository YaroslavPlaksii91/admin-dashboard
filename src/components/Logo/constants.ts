export const LOGO_TYPES = Object.freeze({
  VERTICAL: 'vertical',
  HORIZONTAL: 'horizontal',
});
