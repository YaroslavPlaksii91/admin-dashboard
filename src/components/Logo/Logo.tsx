import { FC } from 'react';
import { Link } from 'react-router-dom';
import { Typography, SvgIcon } from '@mui/material';
import classNames from 'classnames';

import { ROUTES } from '@routes/constants';

import { LOGO_TYPES } from './constants';
import { LogoProps } from './types';

import styles from './Logo.module.css';

export const Logo: FC<LogoProps> = ({ type = LOGO_TYPES.VERTICAL }) => {
  const wrapperClassName = classNames({
    [styles.wrapperVertical]: type === LOGO_TYPES.VERTICAL,
    [styles.wrapperHorizontal]: type === LOGO_TYPES.HORIZONTAL,
  });

  const iconClassName = classNames({
    [styles.iconVertical]: type === LOGO_TYPES.VERTICAL,
    [styles.iconHorizontal]: type === LOGO_TYPES.HORIZONTAL,
  });

  return (
    <div className={wrapperClassName}>
      <Link to={ROUTES.HOME_PAGE} className={styles.link}>
        <SvgIcon titleAccess="Logo" className={iconClassName}>
          <use href="/src/assets/icons/sprite.svg#icon-logo"></use>
        </SvgIcon>
        <Typography
          variant="h3"
          component="p"
          sx={{ opacity: 0.7, color: 'grayLightColor' }}
        >
          Dashboard Kit
        </Typography>
      </Link>
    </div>
  );
};
