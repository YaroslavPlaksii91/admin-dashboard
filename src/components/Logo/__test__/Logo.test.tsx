import { MemoryRouter } from 'react-router-dom';
import { cleanup, render, screen } from '@testing-library/react';

import { LOGO_TYPES } from '../constants';
import { Logo } from '../Logo';

describe('Logo component', () => {
  afterEach(cleanup);

  it('should render Logo component with vertical type', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Logo component with horizontal type', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Logo type={LOGO_TYPES.HORIZONTAL} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be in the document', () => {
    render(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>,
    );

    const logoIcon = screen.getByRole('img');
    const logoText = screen.getByText('Dashboard Kit');

    expect(logoIcon).toBeInTheDocument();
    expect(logoText).toBeInTheDocument();
  });
});
