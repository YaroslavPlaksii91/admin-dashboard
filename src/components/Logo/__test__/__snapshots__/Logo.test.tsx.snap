// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render Logo component with horizontal type 1`] = `
<DocumentFragment>
  <div
    class="undefined"
  >
    <a
      href="/"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium undefined css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-logo"
        />
        <title>
          Logo
        </title>
      </svg>
      <p
        class="MuiTypography-root MuiTypography-h3 css-army73-MuiTypography-root"
      >
        Dashboard Kit
      </p>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Logo component should render Logo component with vertical type 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <a
      href="/"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-logo"
        />
        <title>
          Logo
        </title>
      </svg>
      <p
        class="MuiTypography-root MuiTypography-h3 css-army73-MuiTypography-root"
      >
        Dashboard Kit
      </p>
    </a>
  </div>
</DocumentFragment>
`;
